src/main/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSender.java

```diff
@@ -1,25 +1,140 @@
 import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.HttpRequestFailedException;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.MissingConfigurationException;
+import com.ericsson.eiffelcommons.utils.HttpRequest.HttpMethod;
 import com.google.gson.Gson;
-import com.google.gson.JsonObject;
 
 public class EiffelEventSender {
+    private static final String GENERATE_PUBLISH_ENDPOINT = "/generateAndPublish/";
+    private static final String MESSAGE_PROTOCOL = "mp";
+    private static final String MESSAGE_TYPE = "msgType";
+    private static final String EIFFEL_PROTOCOL = "eiffelsemantics";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EiffelEventSender.class);
 
     private final Gson gson = new Gson();
-    private final EiffelPluginConfiguration pluginConfig;
+    private String eiffelMessage;
+    private String eiffelType;
+    private EiffelPluginConfiguration pluginConfig;
+    private HttpRequest httpRequest;
 
     public EiffelEventSender(EiffelPluginConfiguration pluginConfig) {
+        this(pluginConfig, new HttpRequest(HttpMethod.POST));
+    }
+
+    public EiffelEventSender(EiffelPluginConfiguration pluginConfig, HttpRequest httpRequest) {
+        this.httpRequest = httpRequest;
         this.pluginConfig = pluginConfig;
     }
 
-    public void send(EiffelEvent eiffelEvent) {
-        JsonObject eiffelEventJson = gson.toJsonTree(eiffelEvent).getAsJsonObject();
-        System.out.println("Event to send::: " + eiffelEventJson.toString());
-        System.out.println("REMReM URL ::: " + pluginConfig.getRemremPublishURL());
-        // TODO Implement event sending using REMReM details from pluginconfig.
+    /**
+     * Sends a REMReM eiffel message using
+     *
+     */
+    public void send() {
+        try {
+            verifyConfiguration();
+            generateAndPublish();
+        } catch (URISyntaxException | IOException | MissingConfigurationException
+                | HttpRequestFailedException e) {
+            LOGGER.error("Failed to send eiffel message.", e);
+        }
+    }
+
+    public EiffelEventSender setMessage(final EiffelEvent eiffelMessage) {
+        this.eiffelMessage = gson.toJsonTree(eiffelMessage).toString();
+        return this;
+    }
+
+    public EiffelEventSender setType(final String eiffelType) {
+        this.eiffelType = eiffelType;
+        return this;
+    }
+
+    private void generateAndPublish()
+            throws URISyntaxException, IOException, MissingConfigurationException,
+            HttpRequestFailedException {
+        assembleRequest();
+        final ResponseEntity response = httpRequest.performRequest();
+        verifyResponse(response);
     }
 
+    private void verifyResponse(ResponseEntity response) throws HttpRequestFailedException {
```

Comment:
Do we want to have retries?

Reply:
Well I can only speak for myself but I think that REMReM should be responsible to retries. I don't think it currently does this but it should be part of that service.

Reply:
Will not work if REMReM is down :)

Reply:
I see, I assumed you were talking about sending the message out to the mb. There is something called HttpRequestRetryHandler in apache http lib that might be useful for failed requests. This would require changes in eiffel-commons though. My only apprehension is that I'm not sure what the benefit of using remrem is then...

Reply:
Is remrem down, we won't be able to do much. Although, we can't know if its simply down or some other temporary network issue occured, i think we should have some kind of retry

Reply:
This implementation should a configurable option in HttpRequest then. I can create an issue on this so that the scope does not grow on this one.

Reply:
Issue https://github.com/eiffel-community/eiffel-commons/issues/29